packages/typespec-client-generator-core/test/tester.ts

```diff
@@ -0,0 +1,120 @@
+import { Entity, Program, resolvePath } from "@typespec/compiler";
+import {
+  TestCompileResult,
+  TesterInstance,
+  createTester,
+  resolveVirtualPath,
+} from "@typespec/compiler/testing";
+import { CreateSdkContextOptions, createSdkContext } from "../src/context.js";
+import { SdkContext, SdkHttpOperation, SdkServiceOperation } from "../src/interfaces.js";
+import { BrandedSdkEmitterOptionsInterface } from "../src/internal-utils.js";
+
+export interface SdkTesterOptions extends BrandedSdkEmitterOptionsInterface {
+  emitterName?: string;
+  autoImports?: string[];
+  autoUsings?: string[];
+  packageName?: string;
+}
+
+export const TcgcTester = createTester(resolvePath(import.meta.dirname, ".."), {
+  libraries: [
+    "@typespec/http",
+    "@typespec/rest",
+    "@typespec/versioning",
+    "@azure-tools/typespec-client-generator-core",
+  ],
+})
+  .importLibraries()
+  .using("Http", "Rest", "Versioning", "Azure.ClientGenerator.Core");
+
+export interface TcgcTesterInstance extends TesterInstance {
```

Comment:
@timotheeguerin Is this still a good way to do the context injection for the library, also create reusable tsp content structure in tester v2?

Reply:
anyway, since TCGC is not exactly an emitter, I use another helper to create context.